OpenRA.Game/Game.cs

```diff
@@ -776,9 +784,16 @@ static void Loop()
 
 			while (state == RunStatus.Running)
 			{
-				// Ideal time between logic updates. Timestep = 0 means the game is paused
-				// but we still call LogicTick() because it handles pausing internally.
-				var logicInterval = worldRenderer != null && worldRenderer.World.Timestep != 0 ? worldRenderer.World.Timestep : Ui.Timestep;
+				// Ideal time between logic updates.
+				var logicInterval = Ui.Timestep;
+				if (worldRenderer != null)
+				{
+					var world = worldRenderer.World;
+					logicInterval = world == null ? Ui.Timestep :
+						world.IsLoadingGameSave ? 1 :
+						world.IsReplay ? world.ReplayTimestep :
```

Comment:
This should probably be `world.IsReplay && world.ReplayTimestep != 0` to avoid problems with paused replays.

Reply:
IMO it would be a good idea to keep the comment about `Timestep = 0 means the game is paused` too.